specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-07-01/computeRPCommon.json

```diff
@@ -1913,6 +1913,89 @@
         }
       },
       "description": "Specifies additional settings to be applied when patch mode AutomaticByPlatform is selected in Linux patch settings."
+    },
+    "AttachDetachDataDisksRequest": {
+      "properties": {
+        "attachDataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AttachDataDisk"
+          },
+          "description": "The list of managed data disks to be attached."
+        },
+        "detachDataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DetachDataDisk"
+          },
+          "description": "The list of managed data disks to be detached."
+        }
+      },
+      "description": "Specifies the input for attaching and detaching a list of managed data disks."
+    },
+    "AttachDetachDataDisksResponse": {
+      "properties": {
+        "attachedDataDisks": {
```

Comment:
> **[Query]** Why only attached disk in response ? What would be response when just detachDataDisks is passed in the request ? Is it appropriate to have blank body in that case .

In case of only detach, we need to return IDs again and it wont different from what is sent in the request. Hence, thought of keeping it empty. Please suggest.

Reply:
[**suggestion 1**]: We might say that we would return only list of disk's attached disk to the VM (consisting of both input disks and the already attached disks) and case of detach we can return empty response. 

[**suggestion 2**]: We can add seperate key named "detachedDataDisks" which will contain requested disks to be detached and "attachedDataDisks"| will contains requested data disks to be attached.

I am aligned with both these suggestions, in favour a bit to second one, because the customer gets the reference of what operations were performed on which disks.